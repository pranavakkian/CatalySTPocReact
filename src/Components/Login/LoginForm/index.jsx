import PropTypes from 'prop-types';
import { useState } from 'react';
import { Button, Form } from 'semantic-ui-react';

const LoginForm = (props) => {
  const { handleLoginForm } = props;
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  return (
    <div>
      <Form>
        <Form.Field>
          <input id="username" placeholder="Username" label="Username" value={username} onChange={(e) => setUsername(e.target.value)} />
        </Form.Field>
        <Form.Field>
          <input placeholder="Password" value={password} label="Password" onChange={(e) => setPassword(e.target.value)} />
        </Form.Field>
        <Button type="submit" onClick={() => handleLoginForm(username, password)}>Submit</Button>
      </Form>
    </div>
  );
};

LoginForm.propTypes = {
  handleLoginForm: PropTypes.func.isRequired,
};
export default LoginForm;
