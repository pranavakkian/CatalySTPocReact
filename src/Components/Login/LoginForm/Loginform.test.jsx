import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LoginForm from '.';

describe('LoginForm', () => {
  const mock = jest.fn();
  test('Component rendered successfully', () => {
    const { container } = render(<LoginForm handleLoginForm={mock} />);
    expect(container).toMatchSnapshot();
  });
  test('handles username change correctly', () => {
    render(<LoginForm handleLoginForm={mock} />);
    expect(screen.getAllByRole('textbox')[0]).toHaveValue('');
    userEvent.type(screen.getAllByRole('textbox')[0], 'abc123');
    expect(screen.getAllByRole('textbox')[0]).toHaveValue('abc123');
  });
  test('handles password change correctly', () => {
    render(<LoginForm handleLoginForm={mock} />);
    expect(screen.getAllByRole('textbox')[1]).toHaveValue('');
    userEvent.type(screen.getAllByRole('textbox')[1], 'abc123');
    expect(screen.getAllByRole('textbox')[1]).toHaveValue('abc123');
  });
  test('Submit Button Working Properly', () => {
    render(<LoginForm handleLoginForm={mock} />);
    userEvent.type(screen.getAllByRole('textbox')[0], 'admin');
    userEvent.type(screen.getAllByRole('textbox')[1], 'admin');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(mock).toHaveBeenNthCalledWith(1, 'admin', 'admin');
  });
  test('Submit Button Working Properly', () => {
    render(<LoginForm handleLoginForm={mock} />);
    userEvent.type(screen.getAllByRole('textbox')[0], 'abc');
    userEvent.type(screen.getAllByRole('textbox')[1], 'abc123');
    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
    expect(mock).toHaveBeenNthCalledWith(1, 'abc', 'abc123');
  });
});
