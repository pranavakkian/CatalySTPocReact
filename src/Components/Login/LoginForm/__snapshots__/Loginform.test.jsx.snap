// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component rendered successfully 1`] = `
<div>
  <div>
    <form
      class="ui form"
    >
      <div
        class="field"
      >
        <div
          class="ui input"
        >
          <input
            id="username"
            label="Username"
            placeholder="Username"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <input
          label="Password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="ui button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
